### PR TITLE
Fixed UserIndex::rowOf() function for listener proxies

### PR DESCRIPTION
**Pull request by Krzmbrzl**
The ModelIndex::rowOf functions did not take into account that checking
whether ModelIndex::pUser is not sufficient anymore as listener proxies
share the same associated user. Thus to uniquely identify the index for
the actual user, it also has to be taken into account whether the index
at hand corresponds to a listener proxy.

This mis-behavior was responsible for a segfault triggered by being in a
channel you also have a listener proxy in and then being dragged out of
that channel (the dragging part is important). During this process it
would delete the wrong item which it later wanted to select again
(causing access to deleted memory) resulting in the segfault.

### CODE REVIEW

File: src/mumble/UserModel.cpp [hunk 1]
```diff
@@ -990,6 +990,7 @@ void UserModel::removeUser(ClientUser *p) {
 	updateOverlay();
 
 	delete p;
+
```
Comment (davidebeatrici) [inline src/mumble/UserModel.cpp:993]: Unrelated change, do you want to keep it?

File: src/mumble/UserModel.cpp [hunk 2]
```diff
@@ -136,9 +136,9 @@ int ModelItem::rowOf(Channel *c) const {
 	return -1;
 }
 
-int ModelItem::rowOf(ClientUser *p) const {
+int ModelItem::rowOf(ClientUser *p, bool isListener) const {
```
Comment (davidebeatrici) [inline src/mumble/UserModel.cpp:139]:
```suggestion
int ModelItem::rowOf(ClientUser *p, const bool isListener) const {
```

File: src/mumble/UserModel.cpp (hunk 1 above)
Comment (Krzmbrzl) [reply src/mumble/UserModel.cpp:993]: No. That's just a remnant of my debugging statements xD
